c++/src/H5Attribute.cpp

```diff
@@ -551,7 +551,7 @@ Attribute::p_read_variable_len(const DataType &mem_type, H5std_string &strg) con
 
     // Get string from the C char* and release resource allocated by C API
     strg = strg_C;
-    HDfree(strg_C);
+    free(strg_C);
```

Comment:
delete?

Reply:
Same here. You can't mix malloc/free and new/delete.

Reply:
I know.  It was an accident, but you fixed it by changing HDfree to free, and I asked should it be delete?

Reply:
It shouldn't be delete[]. If you expand the code, you'll see that srgc_C is an unallocated char pointer. The memory is allocated by the library and should be freed using free().